benchmark/Streamly/Benchmark/Data/Unfold.hs

```diff
@@ -131,57 +410,122 @@ zipWithM size start =
+          -- , benchIO "singletonM" $ singletonM size
+          -- , benchIO "singleton" $ singleton size
+          -- , benchIO "identity" $ identity size
+          -- , benchIO "const" $ const size
+          , benchIO "unfoldrM" $ unfoldrM size
+          , benchIO "fromList" $ fromList size
+          , benchIO "fromListM" $ fromListM size
+          -- Unimplemented
+          -- , benchIO "fromSVar" $ fromSVar size
+          -- , benchIO "fromProducer" $ fromProducer size
+          ]
     ]
 
-o_1_space_serial_combination :: Int -> [Benchmark]
-o_1_space_serial_combination size =
-    [ bgroup "combination"
-        [ benchIO "zipWithM" $ zipWithM size
-        ]
+
+o_1_space_generation_specialized :: Int -> [Benchmark]
+o_1_space_generation_specialized size =
+    [ bgroup
+          "generation/specialized"
+          [ benchIO "replicateM" $ replicateM size
+          , benchIO "repeatM" $ repeatM size
+          , benchIO "iterateM" $ iterateM size
+          , benchIO "fromIndicesM" $ fromIndicesM size
+          , benchIO "enumerateFromStepIntegral" $ enumerateFromStepIntegral size
+          , benchIO "enumerateFromToIntegral" $ enumerateFromToIntegral size
+          , benchIO "enumerateFromIntegral" $ enumerateFromIntegral size
+          , benchIO "enumerateFromStepNum" $ enumerateFromStepNum size
+          , benchIO "numFrom" $ numFrom size
+          , benchIO "enumerateFromToFractional" $ enumerateFromToFractional size
+          ]
     ]
 
-o_1_space_serial_outerProduct :: Int -> [Benchmark]
-o_1_space_serial_outerProduct size =
-    [ bgroup "outer-product"
-        [ benchIO "toNull" $ toNull size
-        , benchIO "toNull3" $ toNull3 size
-        , benchIO "concat" $ concat size
-        , benchIO "breakAfterSome" $ breakAfterSome size
-        ]
+o_1_space_transformation :: Int -> [Benchmark]
+o_1_space_transformation size =
+    [ bgroup
+          "transformation"
+          [ benchIO "map" $ map size
+          , benchIO "mapM" $ mapM size
+          , benchIO "mapMWithInput" $ mapMWithInput size
+          , benchIO "takeWhileM" $ takeWhileM size
+          , benchIO "takeWhile" $ takeWhile size
+          , benchIO "take" $ take size
+          , benchIO "filter" $ filter size
+          , benchIO "filterM" $ filterM size
+          -- Very small benchmark, reporting in ns
+          -- , benchIO "dropOne" $ dropOne size
+          , benchIO "dropAll" $ dropAll size
+          , benchIO "dropWhileTrue" $ dropWhileTrue size
+          , benchIO "dropWhileFalse" $ dropWhileFalse size
+          , benchIO "dropWhileMTrue" $ dropWhileMTrue size
+          , benchIO "dropWhileMFalse" $ dropWhileMFalse size
+          ]
     ]
 
-o_n_space_serial :: Int -> [Benchmark]
-o_n_space_serial size =
-    [ bgroup "outer-product"
-        [ benchIO "toList" $ toList size
-        , benchIO "toListSome" $ toListSome size
-        ]
+o_1_space_combination :: Int -> [Benchmark]
+o_1_space_combination size =
```

Comment:
We can call this `o_1_space_zip`, zip is always a separate class in all other benchmarks.